src/components/pages/pages.component.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+import { compact, assign } from 'lodash';
+import { withTheme } from 'styled-components';
+import tagComponent from '../../utils/helpers/tags';
+import Page from './page/page.component';
+import {
+  PagesWrapperStyle,
+  PagesContent
+} from './pages.style';
+import baseTheme from '../../style/themes/base';
+
+const NEXT = 'next';
+const PREVIOUS = 'previous';
+const TRANSITION_TIME = 250;
+
+class Pages extends React.Component {
+  constructor(...args) {
+    super(...args);
+
+    /** Direction of animation */
+    this.transitionDirection = NEXT;
+    this.numOfPages = this.numOfPages.bind(this);
```

Comment:
If we don't need special scope of `this` we can turn the functions to be the arrow ones and then remove binding this from the `constructor`